analyzers/src/SonarAnalyzer.Common/Rules/InsteadOfAnyBase.cs

```diff
@@ -124,4 +130,23 @@ protected bool HasValidInvocationOperands(TInvocationExpression invocation, stri
 
     private void RaiseIssue(SonarSyntaxNodeReportingContext c, SyntaxNode invocation, DiagnosticDescriptor rule, string methodName) =>
         c.ReportIssue(Diagnostic.Create(rule, Language.Syntax.NodeIdentifier(invocation)?.GetLocation(), methodName));
+
+    // See https://github.com/SonarSource/sonar-dotnet/issues/7286
+    private bool IsUsedByEntityFramework(SyntaxNode node, SemanticModel model)
```

Comment:
Use `IsInExpressionTree` instead. This applies not just to EF as expression trees are unlikely to be executed in general.